src/components/AddPresetAsset.tsx

```diff
@@ -0,0 +1,180 @@
+import { useEffect, useState } from "react";
+import { useHistory } from "react-router-dom";
+import {
+  Button,
+  Checkbox,
+  Heading2,
+  InfoBlock,
+  InfoBlockVariant,
+  Loader,
+} from "@stellar/design-system";
+import { TextLink } from "components/TextLink";
+import { presetAsset } from "constants/presetAssets";
+import { getErrorMessage } from "helpers/getErrorMessage";
+import { getNetworkConfig } from "helpers/getNetworkConfig";
+import { getPresetAssets } from "helpers/getPresetAssets";
+import { getValidatedUntrustedAsset } from "helpers/getValidatedUntrustedAsset";
+import { log } from "helpers/log";
+import { searchParam } from "helpers/searchParam";
+import { useRedux } from "hooks/useRedux";
+import { ActionStatus, SearchParams } from "types/types.d";
+import { shortenStellarKey } from "helpers/shortenStellarKey";
+
+export const AddPresetAsset = ({ onClose }: { onClose: () => void }) => {
+  const { account, allAssets, settings, untrustedAssets } = useRedux(
+    "account",
+    "allAssets",
+    "settings",
+    "untrustedAssets",
+  );
+  const [presetAssets, setPresetAssets] = useState<presetAsset[]>([]);
+  const [checkedAssets, setCheckedAssets] = useState<{
+    [key: string]: boolean;
+  }>({});
+  const [isValidating, setIsValidating] = useState(false);
+  const [errorMessage, setErrorMessage] = useState("");
+
+  const history = useHistory();
+
+  useEffect(() => {
+    setPresetAssets(getPresetAssets(allAssets.data));
+  }, [allAssets]);
+
+  useEffect(() => {
+    if (untrustedAssets.status === ActionStatus.SUCCESS) {
+      onClose();
+    }
+
+    if (untrustedAssets.errorString) {
+      setErrorMessage(untrustedAssets.errorString);
+    }
+  }, [onClose, untrustedAssets.errorString, untrustedAssets.status]);
+
+  const getAssetId = (asset: presetAsset) =>
+    `${asset.assetCode}:${asset.homeDomain || asset.issuerPublicKey}`;
+
+  const handleOnChangeCheckbox = (asset: presetAsset) => {
+    const updatedCheckedAssets = JSON.parse(JSON.stringify(checkedAssets));
+    const assetId = getAssetId(asset);
+    updatedCheckedAssets[assetId] = !checkedAssets[assetId];
+    setCheckedAssets(updatedCheckedAssets);
+  };
+
+  const handleAddUntrustedAssets = async (assetList: presetAsset[]) => {
+    setErrorMessage("");
+    setIsValidating(true);
+
+    try {
+      const validatedAssetValues = await Promise.all(
+        assetList.map(async (pAsset) => {
+          const { assetCode, issuerPublicKey, homeDomain } = pAsset;
+
+          if (!(homeDomain || issuerPublicKey)) {
+            const errorMsg = `Home domain OR issuer public key is required with asset code ${assetCode}`;
+            log.error({ title: errorMsg });
+            throw new Error(errorMsg);
```

Comment:
We might not need `log.error({ title: errorMsg });` here if we're logging it in `catch`.

Reply:
Good point! Fixed in https://github.com/stellar/stellar-demo-wallet/pull/206/commits/3a63377862a8265c9bc0e413fcd018ed7a13d9f0.